module/documents/activity/attack.mjs

```diff
@@ -177,14 +184,21 @@ export default class AttackActivity extends ActivityMixin(AttackActivityData) {
     const roll = await d20Roll(oldRollConfig);
     if ( roll === null ) return;
 
+    const ammo = this.actor?.items.get(roll.options.ammunition);
+    let ammoUpdate = null;
+    if ( ammo ) {
+      ammoUpdate = { id: ammo.id, quantity: Math.max(0, ammo.system.quantity - 1) };
+      ammoUpdate.destroy = ammo.system.uses.autoDestroy && (ammoUpdate.quantity === 0);
```

Comment:
Ammo isn't currently destroyed when the last of it is spent, and that'd be pretty annoying when a ranged character has to refill all the time.

Reply:
Well, it would only be deleted if you have "Destroy on Empty" set on the item, which it isn't for any of the SRD ammo.

Reply:
Probably need a different hint for ammo consumables only, since I think the normal `autoDestroy` consumable behaviour is to reduce the quantity by 1 when charges are used up (and then delete it at 0). But for ammo we ignore charges and always reduce the quantity by 1, but only destroy it at 0 if `autoDestroy` is set, which is fairly different behaviour.

Reply:
I've been thinking we need two separate options. One allows reducing the quantity when uses are used up, and the second for destroying the item when its quantity reaches zero from consumption. That would allow for situations where you want to just have the quantity hit `0` by using uses but keep the item on the sheet.

Reply:
I think the default behaviour we have currently is sensible, we just need a different hint to explain it. We can't add configuration for every niche use case and preference, otherwise the UI gets overwhelming.